pkg/hubble/parser/seven/parser.go

```diff
@@ -280,8 +280,8 @@ func decodeEndpoint(endpoint accesslog.EndpointInfo, namespace, podName string)
 	labels := endpoint.Labels
 	sort.Strings(labels)
 	return &pb.Endpoint{
-		ID:        endpoint.ID,
-		Identity:  endpoint.Identity,
+		ID:        uint32(endpoint.ID),
```

Comment:
Does anyone know why `pkg.proxy.accesslog.EndpointInfo.[ID,Identity]` are `uint64` ? These type casts don't look safe to me.

Reply:
See https://github.com/cilium/cilium/pull/11161#issuecomment-619830199 for the rationale - the values will never exceed 32 bit (or 16 bit in this case).

Reply:
But these two instances are from Cilium's proxy package hence not something that falls into your description, no?

Reply:
True, but they suffered a similar fate: `EndpointInfo` was intended for JSON serialization (see  https://github.com/cilium/cilium/pull/964), therefore I assume the type definitions were generated from some JSON-to-Go-struct tool, which picked `int64` for the type.

I've checked all initialization of `EndpointInfo.ID` and they all end take the value taken from `GetID` here (also added by the above PR) which casts a 16 bit integer to 64 bit
https://github.com/cilium/cilium/blob/dd67d2d37f1a07ed5eeaff68cffd09212de62910/pkg/endpoint/endpoint.go#L429

So I don't see any safety issue here.